app/services/file_thawer.py

```
@@ -48,11 +48,53 @@ def thaw_file(
+                    except Exception:
+                        # Clean up temp file if decryption failed
+                        if target_path.exists():
+                            target_path.unlink()
+                        raise
+
+                    # Remove encrypted file if operation was MOVE or SYMLINK (assuming symlink targeted the encrypted file)
+                    # For COPY, we might want to keep it? The logic below handles unlinking for MOVE.
+                    if file_record.operation_type.value != "copy":
+                        cold_path.unlink()
+
+                except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)